src/Path/Surface.hs

```diff
@@ -1,76 +1,45 @@
-{-# LANGUAGE DeriveTraversable, FlexibleInstances, LambdaCase, MultiParamTypeClasses #-}
+{-# LANGUAGE DeriveTraversable, FlexibleInstances, LambdaCase, MultiParamTypeClasses, TypeOperators #-}
 module Path.Surface where
 
+import Path.Core
 import Path.Name
-import Path.Subst
 import Path.Term
 import Path.Usage
 
-data Surface v a
-  = Var v
-  | Lam Name a
-  | a :@ a
-  | Type
-  | Pi Name Usage a a
-  | ForAll Name a a
-  | Hole v
-  | Implicit
+type Surface b v = Sugar b :+: Implicit v :+: Core b v
+
+data Sugar b a
+  = ForAll b a a
+  | (Usage, a) :-> a
```

Comment:
We now provide explicit syntax sugar for (non-dependent) function arrows. And it doesn’t require us to extend the elaborator!